### PR TITLE
#162170721 Users should be able to retrieve all profiles

### PR DESCRIPTION
**Pull request by oluwajuwon**
#### What does this PR do?
This PR handles retrieving a list of all user's profiles

#### Description of Task to be completed?
Have the following endpoint working
GET /api/profiles
- Write tests for this  endpoint
- Create controller for retrieving all user profiles except logged in user
- User should see a list of author's profiles on the platform with the number of articles they have
- Document this API with Swagger API Doc

#### How should this be manually tested?
- Clone this repo
- In your terminal cd into the folder it creates
- Run **npm install** to install all the dependencies that will be used by the project
- Check the **.env.sample** to create a **.env** file in your root folder
- Run **npm run dev:start** to start the app
- Signup through **localhost:3000/api/users/signup** or Login through **localhost:3000/api/users/login** respectively using postman as a client
- Add your generated token as a header
- Send a GET request to **localhost:3000/api/profiles** to retrieve all user's profiles
- Run **npm test** to test the app
- View the Swagger UI on **localhost:3000/api/docs**

#### Any background context you want to provide?
- Authentication is required
- Only an authenticated user can the list of all the profiles

#### What are the relevant pivotal tracker stories? (If applicable)
[#162170721](https://www.pivotaltracker.com/story/show/162170721)

#### Screenshots (If applicable)

screen shot 2018-12-14 at 5 04 02 am

#### Questions:

### CODE REVIEW

File: src/controllers/userProfileHandler.js [hunk 1]
```diff
@@ -130,8 +136,73 @@ class UserProfileController {
...
+   * @memberof UserProfileController
+   */
+  static async getAllProfiles(request, response) {
+    const { payload } = request.userData;
+    const { username } = payload;
+    try {
+      let allProfiles = await User.findAll({
+        where: {
+          username: {
+            $notLike: `${username}`
```
Comment (okoroemeka) [inline src/controllers/userProfileHandler.js:158]: What does `$notLike` do in the query?

**Review by okoroemeka — APPROVED** [review]

File: src/controllers/userProfileHandler.js (hunk 1 above)
Comment (oluwajuwon) [reply src/controllers/userProfileHandler.js:158]: @okoroemeka nice observation, I'm using that return just the users with usernames that aren't like the username of the logged in user. So it returns every other user profile except the person logged in

Comment (frostyblok) [reply src/controllers/userProfileHandler.js:158]: Nice explanation, Juwon.

**Review by frostyblok — APPROVED** [review]
Hi Juwon, nice job you have done here. I like how you were able to implement this feature while writing elegant code;  you were able to use 'map' you loop through an array and return an array as against 'forEach'. I have learned a lot from your code. Good job!

**Review by akhilome — APPROVED** [review]
Kudos on delivering this feature in a timely fashion, @oluwajuwon. I love the fact that you took the pain to change some of the file names to conform to the naming convention the TTL suggested.

**Pull request merged**